trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,401 @@
+"""
+Default value of the kernel variance parameter for classification models.
+"""
+
+
+MAX_NUM_INDUCING_POINTS = tf.cast(500, dtype=tf.int32)
+"""
+Default maximum number of inducing points.
+"""
+
+
+NUM_INDUCING_POINTS_PER_DIM = tf.cast(25, dtype=tf.int32)
+"""
+Default number of inducing points per dimension of the search space.
+"""
+
+
+SNR_LIKELIHOOD = tf.cast(10, dtype=gpflow.default_float())
+"""
+Default value used for initializing (noise) variance parameter of the likelihood function.
+If user does not specify it, the noise variance is set to maintain the signal to noise ratio
+determined by this default value. Signal variance in the kernel is set to the empirical variance.
+"""
+
+
+def build_gpr(
+    data: Dataset,
+    search_space: SearchSpace,
+    kernel_priors: bool = True,
+    likelihood_variance: Optional[float] = None,
+    trainable_likelihood: bool = False,
+) -> GPR:
+    """
+    Build a :class:`~gpflow.models.GPR` model with sensible initial parameters and
+    priors. We use :class:`~gpflow.kernels.Matern52` kernel and
+    :class:`~gpflow.mean_functions.Constant` mean function in the model as we found this combination
+    to be effective in most settings.
+
+    We set priors for kernel hyperparameters by default in order to stabilize model fitting. We
+    found the priors below to be highly effective for objective functions defined over the unit
+    hypercube. They do seem to work for other search space sizes, but we advise caution when using
+    them in such search spaces. Using priors allows for using maximum a posteriori estimate of
+    these kernel parameters during model fitting.
+
+    Note that although we scale parameters as a function of the size of the search space, ideally
+    inputs should be normalised to the unit hypercube before building a model.
+
+    :param data: Dataset from the initial design, used for estimating the variance of observations.
+    :param search_space: Search space for performing Bayesian optimization, used for scaling the
+        parameters.
+    :param kernel_priors: If set to `True` (default) priors are set for kernel parameters (variance
+        and lengthscale).
+    :param likelihood_variance: Likelihood (noise) variance parameter can be optionally set to a
+        certain value. If left unspecified (default), the noise variance is set to maintain the
+        signal to noise ratio of value given by ``SNR_LIKELIHOOD``, where signal variance in the
+        kernel is set to the empirical variance.
+    :param trainable_likelihood: If set to `True` Gaussian likelihood parameter is set to
+        non-trainable. By default set to `False`.
+    :return: A :class:`~gpflow.models.GPR` model.
+    """
+    empirical_mean, empirical_variance, _ = _get_data_stats(data)
+
+    kernel = _get_kernel(empirical_variance, search_space, kernel_priors, kernel_priors)
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    if likelihood_variance is None:
```

Comment:
Not sure if it's worth it, but you could also factor out `_get_noise_variance(likelihood_variance, empirical_variance)`.